plugins/inputs/statsd/statsd.go

```diff
@@ -735,6 +751,15 @@ func parseKeyValue(keyvalue string) (string, string) {
 // Delete* options, because those are dealt with in the Gather function.
 func (s *Statsd) aggregate(m metric) {
 	switch m.mtype {
+	case "d":
+		fields := make(map[string]interface{})
+		fields[m.field] = m.floatvalue
+		cached := cacheddistributions{
+			name:   m.name,
+			fields: fields,
+			tags:   m.tags,
+		}
```

Comment:
You are always setting just one field, why do you use a map for this? You could also store the name and value instead of allocating over and over.

Reply:
So I updated the `cacheddistributions` interface and replaced field with value. The repeated allocation now happens in the `Gather` method where the accumulator's `addFields` requires a map argument.